File: taurus-protect-sdk-go/test/integration/users_test.go
package integration

import (
	"context"
	"testing"

	"github.com/taurushq-io/taurus-protect-sdk/taurus-protect-sdk-go/pkg/protect/model"
)

func TestIntegration_GetMe(t *testing.T) {
	skipIfNotIntegration(t)
	client := getTestClient(t)
	defer client.Close()

	ctx := context.Background()
	user, err := client.Users().GetMe(ctx)
	if err != nil {
		t.Fatalf("GetMe() error = %v", err)
	}

	t.Logf("Current user:")
	t.Logf("  ID: %s", user.ID)
	t.Logf("  Username: %s", user.Username)
	t.Logf("  Email: %s", user.Email)
	t.Logf("  Status: %s", user.Status)
}

func TestIntegration_ListUsers(t *testing.T) {
	skipIfNotIntegration(t)
	client := getTestClient(t)
	defer client.Close()

	ctx := context.Background()
	result, err := client.Users().ListUsers(ctx, &model.ListUsersOptions{
		Limit: 10,
	})
	if err != nil {
		t.Fatalf("ListUsers() error = %v", err)
	}

	t.Logf("Found %d users", len(result.Users))
	for _, u := range result.Users {
		t.Logf("User: ID=%s, Username=%s, Email=%s", u.ID, u.Username, u.Email)
	}
}

File: taurus-protect-sdk-typescript/tests/integration/users.test.ts
/**
 * Integration tests for Users API.
 *
 * These tests require a live API connection. Configure via environment variables
 * or hard-coded defaults in helpers.ts.
 *
 * Tests use the high-level UserService (client.users) rather than
 * the raw OpenAPI API to demonstrate proper SDK usage patterns.
 */

import { skipIfNotIntegration, getTestClient } from "./helpers";

describe("Integration: Users", () => {
  beforeEach(() => {
    try {
      skipIfNotIntegration();
    } catch (e) {
      if (e instanceof Error && e.message === "SKIP_INTEGRATION") {
        return;
      }
      throw e;
    }
  });

  it("should get current user (getCurrentUser)", async () => {
    try {
      skipIfNotIntegration();
    } catch (e) {
      if (e instanceof Error && e.message === "SKIP_INTEGRATION") {
        return;
      }
      throw e;
    }

    const client = getTestClient();
    try {
      const user = await client.users.getCurrentUser();

      expect(user).toBeDefined();

      console.log("Current user details:");
      console.log(`  ID: ${user.id}`);
      console.log(`  Email: ${user.email}`);
      console.log(`  Name: ${user.firstName} ${user.lastName}`);
      console.log(`  Status: ${user.status}`);
      console.log(`  Roles: ${user.roles?.join(", ") || "none"}`);
    } finally {
      client.close();
    }
  });

  it("should list users", async () => {
    try {
      skipIfNotIntegration();
    } catch (e) {
      if (e instanceof Error && e.message === "SKIP_INTEGRATION") {
        return;
      }
      throw e;
    }

    const client = getTestClient();
    try {
      const result = await client.users.list({ limit: 10 });

      const users = result.items;
      console.log(`Found ${users.length} users`);

      if (result.pagination) {
        console.log(`Total items: ${result.pagination.totalItems}`);
      }

      for (const user of users) {
        console.log(
          `User: ID=${user.id}, Email=${user.email}, Name=${user.firstName} ${user.lastName}, Status=${user.status}`
        );
      }

      expect(Array.isArray(users)).toBe(true);
    } finally {
      client.close();
    }
  });

  it("should filter users by status", async () => {
    try {
      skipIfNotIntegration();
    } catch (e) {
      if (e instanceof Error && e.message === "SKIP_INTEGRATION") {
        return;
      }
      throw e;
    }

    const client = getTestClient();
    try {
      const result = await client.users.list({
        limit: 10,
        status: "ACTIVE",
      });

      const users = result.items;
      console.log(`Found ${users.length} active users`);

      for (const user of users) {
        console.log(
          `Active User: ID=${user.id}, Email=${user.email}, Status=${user.status}`
        );
      }

      expect(Array.isArray(users)).toBe(true);
      // All returned users should be active
      for (const user of users) {
        expect(user.status).toBe("ACTIVE");
      }
    } catch (e) {
      // Skip if the API doesn't support status filtering (returns 400)
      if (e instanceof Error && e.message.includes("Response returned an error code")) {
        console.log("Skipping: API validation error - status filter may not be supported");
        return;
      }
      throw e;
    } finally {
      client.close();
    }
  });
});
